applications/chats/services.py

```diff
@@ -0,0 +1,35 @@
+from django.core.exceptions import ValidationError
+from django.db import transaction
+from django.utils.translation import gettext_lazy as _
+
+from .models import Room
+
+
+class RoomCreator:
+
+    @transaction.atomic
+    def create_from_target_match(self, target, *members):
+        if self._exists_equal_room(target.topic, *members):
+            return None
+
+        room = Room.objects.create(topic=target.topic)
+
+        self._validate_members(room, *members)
+
+        room.members.add(*members)
+
+    def _exists_equal_room(self, topic, *members):
+        if not len(members) == 2:
+            return False
+
+        return Room.objects.filter(
+            members__id__in=[m.id for m in members],
+            topic=topic,
+        ).exists()
+
+    def _validate_members(self, room, *members):
+        if len(members) < 2:
+            raise ValidationError({'members': _('You must create a room with two members')})
+
+        if len(members) > 2:
+            raise ValidationError({'members': _('You can‘t create a room with more than two members')})
```

Comment:
Looks like the `room` param is not necessary.

Reply:
I think you could do just one if comparing against 2 and say that only 2 members are allowed.
```suggestion
    def _validate_members(self, room, *members):
        if len(members) != 2:
            raise ValidationError({'members': _('You must create a room with two members')})
```